### PR TITLE
14-feature-10-authentication-google

### PR DESCRIPTION
**Pull request by ayeza-zamudio3**
# Description
Authenthication with google

### CODE REVIEW

**Review by GerardoTovar — APPROVED** [review]
looks good, just need to integrate it

**Review by daniel-yocupicio — DISMISSED** [review]

File: App/src/libreria/authenticationGoogle.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  statusCodes,
+} from '@react-native-google-signin/google-signin';
+import auth from '@react-native-firebase/auth';
+import firestore from '@react-native-firebase/firestore';
+//const signIn = (firstname, suscribe, email, password, navigation)
+export const signInGoogle = async navigation => {
+  try {
+    GoogleSignin.configure({
+      webClientId:
+        '1023121760646-4o0p6e0meeogfe9he5poq2llq4oo873e.apps.googleusercontent.com',
```
Comment (YilKanda) [inline App/src/libreria/authenticationGoogle.js:12]: Your Google ID should not appear on GitHub, use environment variables to hide it.

File: App/src/libreria/authenticationGoogle.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    return navigation.navigate('Home');
+  } catch (error) {
+    if (error.code === statusCodes.SIGN_IN_CANCELLED) {
+      console.log('user cancelled the login flow: ' + error.code);
+    } else if (error.code === statusCodes.IN_PROGRESS) {
+      console.log('operation (sign in) is in progress already: ' + error.code);
+    } else if (error.code === statusCodes.PLAY_SERVICES_NOT_AVAILABLE) {
+      console.log('play services not available or outdated: ' + error.code);
+    } else {
+      console.log(error);
```
Comment (YilKanda) [inline App/src/libreria/authenticationGoogle.js:40]: In GitHub, no debugging code should appear, such as console.log()

You can use Alert.alert() instead.

File: App/src/libreria/authenticationGoogle.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  firestore()
+    .collection('Users')
+    .add({
+      firstname: firstname,
+      suscribe: suscribe,
+      email: email,
+      uid: uid,
+    })
+    .then(() => {
+      console.log('Usuario agregado! => Enviando  A Home');
```
Comment (YilKanda) [inline App/src/libreria/authenticationGoogle.js:55]: In GitHub, no debugging code should appear, such as console.log()

You can use Alert.alert() instead.

File: App/src/libreria/authenticationGoogle.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      navigation.navigate('Home');
+    })
+    .catch(error => console.log(error));
```
Comment (YilKanda) [inline App/src/libreria/authenticationGoogle.js:58]: In GitHub, no debugging code should appear, such as console.log()

You can use Alert.alert() instead.

File: App/src/libreria/authenticationGoogle.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import {
+  GoogleSignin,
+  statusCodes,
+} from '@react-native-google-signin/google-signin';
+import auth from '@react-native-firebase/auth';
+import firestore from '@react-native-firebase/firestore';
+//const signIn = (firstname, suscribe, email, password, navigation)
```
Comment (YilKanda) [inline App/src/libreria/authenticationGoogle.js:7]: Code that you do not use, delete it.

**Review by YilKanda — CHANGES_REQUESTED** [review]

File: App/.babelrc [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+        "blocklist": null,
+        "allowlist": null,
+        "blacklist": null, // DEPRECATED
+        "whitelist": null, // DEPRECATED
+        "safe": false,
+        "allowUndefined": true,
+        "verbose": false
+      }]
+    ]
+  }
```
Comment (YilKanda) [inline App/.babelrc:16]: Add a blank line

File: App/android/app/build.gradle [hunk 7]
```diff
@@ -318,3 +318,5 @@ def isNewArchitectureEnabled() {
     // - Set an environment variable `ORG_GRADLE_PROJECT_newArchEnabled=true`
     return project.hasProperty("newArchEnabled") && project.newArchEnabled == "true"
 }
+
+apply plugin: 'com.google.gms.google-services'
```
Comment (YilKanda) [inline App/android/app/build.gradle:322]: Add a blank line

**Review by YilKanda — CHANGES_REQUESTED** [review]
Resolve conflicts that exist in your branch.

**Pull request closed without merging**